testing/runner.ts

```diff
@@ -58,111 +55,180 @@ function partition(
   );
 }
 
+function expandDirectory(dir: string, options: ExpandGlobOptions): WalkInfo[] {
+  return DIR_GLOBS.flatMap((s: string): WalkInfo[] => [
+    ...expandGlobSync(s, { ...options, root: dir })
+  ]);
+}
+
 /**
- * Given list of globs or URLs to include and exclude and root directory return
- * list of file URLs that should be imported for test runner.
+ * Given a list of globs or URLs to include and exclude and a root directory
+ * from which to expand relative globs, return a list of URLs
+ * (file: or remote) that should be imported for the test runner.
  */
-export async function getMatchingUrls(
-  matchPaths: string[],
-  excludePaths: string[],
-  root: string
+export async function findTestModules(
```

Comment:
Nice :+1: I need more time to go through it in detail, but I suggest writing some more unit tests for this method

Reply:
Done. The test module looks a lot better: https://github.com/denoland/deno_std/blob/09e47c704a64b09be10d1681aef2dfd02ebf3392/testing/runner_test.ts.